File: backend/app/schemas/repo_analysis.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field

from app.models.repo_analysis import RepoAnalysisStatus


class RepoAnalysisRead(BaseModel):
    id: int
    project_id: int
    repo_integration_id: int
    status: RepoAnalysisStatus
    summary: str | None
    detected_stack: dict[str, Any]
    detected_files: list[str]
    detected_folders: list[str]
    signals: dict[str, bool]
    warnings: list[str]
    error_message: str | None
    total_files_scanned: int
    analysis_version: str = "codemap_lite_v1"
    insights: dict[str, Any] = Field(default_factory=dict)
    evidence_files: dict[str, list[str]] = Field(default_factory=dict)
    inspected_files: list[str] = Field(default_factory=list)
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/repoAnalysis.ts
export type RepoAnalysisStatus = "completed" | "failed";

export interface RepoAnalysis {
  id: number;
  project_id: number;
  repo_integration_id: number;
  status: RepoAnalysisStatus;
  summary: string | null;
  detected_stack: Record<string, string[]>;
  detected_files: string[];
  detected_folders: string[];
  signals: Record<string, boolean>;
  warnings: string[];
  error_message: string | null;
  total_files_scanned: number;
  analysis_version?: string;
  insights?: {
    runtimes?: string[];
    package_managers?: string[];
    frameworks?: string[];
    commands?: Record<string, string[]>;
    dependency_counts?: { runtime: number; development: number };
    operational_signals?: string[];
  };
  evidence_files?: Record<string, string[]>;
  inspected_files?: string[];
  created_at: string;
}

export interface ProjectRepoAnalysisOverview {
  project_id: number;
  project_name: string;
  project_status: string;
  repo_owner: string | null;
  repo_name: string | null;
  latest_status: RepoAnalysisStatus | null;
  summary: string | null;
  total_files_scanned: number | null;
  analyzed_at: string | null;
}
